src/test/neorados/common_tests.h

```diff
@@ -15,27 +15,141 @@
 #include <string>
 #include <string_view>
 
+#include <boost/asio.hpp>
+#include <boost/asio/awaitable.hpp>
+
+#include <boost/system/error_code.hpp>
+
 #include "include/neorados/RADOS.hpp"
 
 std::string get_temp_pool_name(std::string_view prefix = {});
 
 template<typename CompletionToken>
-auto create_pool(neorados::RADOS& r, std::string_view pname,
-		 CompletionToken&& token)
+auto create_pool(
+  neorados::RADOS& r,
+  std::string pname,
+  CompletionToken&& token)
+{
+  namespace ba = boost::asio;
+  namespace bs = boost::system;
+  auto&& executor = ba::get_associated_executor(token);
+  std::cerr << "In create_pool" << std::endl;
+  return ba::async_compose<CompletionToken, void(bs::error_code, std::int64_t)>
+    (
+      [&executor, &r, &pname](auto& self) {
+	std::cerr << "In create_pool's async_compose" << std::endl;
+	co_spawn(
+	  executor,
+	  [&r, &self, pname]() mutable -> ba::awaitable<void> {
```

Comment:
careful with the `awaitable` template, it defaults to the polymorphic `any_io_executor`:
```c++
template<typename T, typename Executor = any_io_executor>
class awaitable;
```
when we know the real executor type, we should specify it. we saw a nice speedup from switching to explicit executor types in https://github.com/ceph/ceph/pull/43761

Reply:
I'll do an audit of all these once I get everything working and make them more specific.